### PR TITLE
feat(frontmatter): implement getFrontmatterBlock method

### PR DESCRIPTION
**Pull request by P0lip**
Going to be useful for lazy markdown reading in Studio.

### CODE REVIEW

File: src/frontmatter/frontmatter.ts [hunk 1]
```diff
@@ -94,6 +98,33 @@ export class Frontmatter<T extends object = any> implements IFrontmatter<T> {
...
+    if (value.slice(0, index) !== open || value.charAt(index) !== newline) {
+      return;
+    }
+
+    let offset = value.indexOf(close, index);
+
+    while (offset !== -1 && value.charAt(offset - 1) !== newline) {
+      index = offset + close.length;
+      offset = value.indexOf(close, index);
+    }
```
Comment (billiegoose) [inline src/frontmatter/frontmatter.ts]: Would this work?
```suggestion
   const offset = value.indexOf(newline + close, index);
```

Comment (billiegoose) [reply src/frontmatter/frontmatter.ts]: Or honestly... can we just replace `getFrontMatterBlock` with a single regex?

```js
function getFrontmatterBlock (value) {
  let match = value.match(/^(\s*\n)?---.*?\n---\s*/s)
  return match && match[0]
}
```

**Review by billiegoose — APPROVED** [review]
I hate it, lets merge it. Whatever version you like. It's all the same.

Although I can't help but think my regex version will be substantially faster.

**Comment by billiegoose** [comment]
I'm terribly impatient today. This is such a good feature, I'm just gonna merge it while you're sleeping, and if you don't like my change, I'll find a way to make it up to you. We're all gonna die from coronavirus anyway, so I'm living on the edge today!

**Comment by pytlesk4** [comment]
@wmhilton is living his best life.

**Pull request merged**